contrib/quicktag.lua

```diff
@@ -104,12 +104,24 @@ local function abbrevate_tags(t)
   for i=1,qnr do
     local taglength = string.len(t[i])
     if taglength > max_label_length then
-      local max_2 = math.floor(max_label_length/2)
-      quicktag_label[i] = string.sub(t[i],0,max_2).."..."..string.sub(t[i],taglength - max_2)
+
+      s = {}; k = 0
+      for str in string.gmatch(t[i], "([^|]+)") do
+        s[k] = str
+        k = k + 1
+      end
+
+      quicktag_label[i] = ""
+      k = 0
+      while s[k+1] do
+        quicktag_label[i] = quicktag_label[i] .. string.sub(s[k],0,1) .. ".." .. string.sub(s[k],-1) .. "|"
+        k = k + 1
+      end
+      quicktag_label[i] = quicktag_label[i] .. s[k]
+
```

Comment:
That's true, but I think it's more the tree than the tag itself which causes a problem. The tree description (`...|...|...|...|`) becomes long really fast. And, even if you count letters, with a proportional font, all of them have different sizes and you only get a lose statistical connection between the length of a string on screen and the number of letters. One could think about special handling of the tag itself (opposed to tree branches), but I would rather test if the proposed approach isn't enough or even better in terms of readability/recognizability of the tags.